clang/compiler-course/khokhlov_a_l1_v4/khokhlov_a_l1_v4.cpp

```diff
@@ -0,0 +1,108 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "clang/Rewrite/Core/Rewriter.h"
+#include "llvm/Support/raw_ostream.h"
+#include <unordered_map>
+
+namespace {
+class PrefixVisitor final : public clang::RecursiveASTVisitor<PrefixVisitor> {
+  clang::Rewriter &m_rewriter;
+  std::unordered_map<clang::VarDecl *, std::string> rewrittenDecl;
+
+public:
+  explicit PrefixVisitor(clang::ASTContext *context, clang::Rewriter &rewriter)
+      : m_rewriter(rewriter) {}
+
+  bool VisitVarDecl(clang::VarDecl *varDecl) {
+    if (!varDecl)
+      return true;
+
+    std::string prevName = varDecl->getName().str();
+    if (prevName.empty())
+      return true;
+
+    std::string prefix;
+    if (varDecl->isStaticLocal()) {
+      prefix = "static_";
+    } else if (varDecl->hasGlobalStorage()) {
+      if (varDecl->getStorageClass() == clang::SC_Static) {
+        prefix = "static_";
+      } else if (varDecl->getStorageClass() == clang::SC_Extern) {
+        prefix = "extern_";
+      } else {
+        prefix = "global_";
+      }
+    } else if (varDecl->isLocalVarDecl()) {
+      prefix = "local_";
+    }
+
+    if (!prefix.empty()) {
+      std::string newName = prefix + prevName;
+      m_rewriter.ReplaceText(varDecl->getLocation(), prevName.size(), newName);
+      rewrittenDecl[varDecl] = newName;
```

Comment:
Unnecessary copying should be avoided
```suggestion
      rewrittenDecl[varDecl] = std::move(newName);
```

Reply:
fixed